### ISSUE TITLE
Unintuitive keyboard behaviour on navigating the "Tabs" component

### ISSUE THREAD

**Issue by mr-woodapple** [radzenhq/radzen-blazor#2446]
**Is your feature request related to a problem? Please describe.**
Not really a "problem" per se. I'm using the Radzen Blazor kit for a while now and I've had an accessibility test coming back, stating that tabs cannot be changed using the keyboard.

It took me some time to figure out how to navigate between tabs, with selecting the container that surrounds the tabs and then using the left or right arrow.

**Describe the solution you'd like**
I would like to have the functionality changed of how tabs work. I would expect that if you select a tab itself, you can switch between tabs using the arrow keys (left and right) and then select a tab by pressing ENTER.

That is the behavior I was able to find in pretty much any major ui kit, see the links below for reference:

- [Material Design 3 Tabs (the tabs at the top)](https://m3.material.io/components/tabs/overview)
- [MUI Tabs](https://mui.com/material-ui/react-tabs/)
- [React Aria Tabs](https://react-aria.adobe.com/Tabs)

**Describe alternatives you've considered**
None.

**Additional context**
I would be interested to implement the desired solution myself.

**Comment by mr-woodapple** [comment]
Sorry to open this up to another topic, but I found the same issue for me using the `DataGrid`. Take this example, try navigating the table using the TAB key, then change the `IsFilteringEnabled` bool and try again: [Radzen Playground](https://blazor.radzen.com/playground/43485df3-f69d-44b3-849b-7eaf98441956)

With filtering enabled, I can navigate through the tables header, but I can't with filtering disabled. Even with filtering enabled, I have no clear way of seeing how I can change sorting only using keyboard navigation. 
Instead, I have to know that when the focus is set to the entire `DataGrid` (like in the screenshot below), I can use the arrow keys to navigate the column headers (and change the sorting by pressing ENTER.


My preferred solution would be that after focusing the entire table, I can then (by pressing TAB again) navigate over the column header (and if filtering or sorting is set to true have the TAB focus these elements before switching to the next column). But if that's not possible, a workaround would be to highlight the first column element that get's focus when the entire table has received focus (like in the screenshot above). 

<img width="1623" height="570" alt="Image" src="https://github.com/user-attachments/assets/fd00782d-17f3-4bac-839c-b2ef2667af1b" />

**Comment by mr-woodapple** [comment]
Why was this closed? From what I can see when running the code locally, nothing has changed for what I pointed out. Is that intentionally?

**Comment by enchev** [comment]
It was closed by a pull request that fixed keyboard navigation to work as described in the keyboard navigation section of the relevnat component.

**Comment by mr-woodapple** [comment]
I see that! :) 

But my issue was about that I found the current system to be highly unintuitive (and so did an accessibility report on our end) and I wanted to discuss if it's worth implementing changes to it (and potentially do it myself).

**Comment by enchev** [comment]
> I would like to have the functionality changed of how tabs work. I would expect that if you select a tab itself, you can switch between tabs using the arrow keys (left and right) and then select a tab by pressing ENTER.

This is how the component behaves after my pull request was merged. A regression was introduced by our recent accessibility changes (A → BUTTON), which affected both the Tabs and DataGrid components. 
I've just commited additional fix related to the same: https://github.com/radzenhq/radzen-blazor/commit/7d29bc1645154e7dcb78a0fcc591ad1fcdbc05b1

**Comment by mr-woodapple** [comment]
No - the way it works right now is that you select then entire Tab component - like with the content that's in the tab. The `:active` highlight border is around the entire tab content. And in this situation, the user can press the "Arrow Left" or "Arrow Right" button to switch between tabs. But before pressing the first arrow key, it's not even highlighted that this is an option.

I would suggest that the tab logic is updated to this: 
- Press tab -> select the entire content (the `<RadzenTabs/>` element), show border around the entire element
- Press tab -> select the tab header (the `<RadzenTabsItem />` element), show border around the first tab header/nav
   - Press "Arrow Right" -> select the second tab, show border around the second tab header
   - etc.
- Press tab -> select the tab body/content
